src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -0,0 +1,259 @@
+        return filterAndSortProfiles(this.props.molecularProfiles);
+    }
+
+    @computed get hasMutationData() {
+        return !!_.find(
+            this.props.molecularProfiles,
+            profile=>profile.molecularAlterationType === AlterationTypeConstants.MUTATION_EXTENDED
+        );
+    }
+
+    @computed get molecularProfileIdToMolecularProfile() {
+        return _.keyBy(this.profiles, profile=>profile.molecularProfileId);
+    }
+
+    private coExpressionCache:CoExpressionCache = new CoExpressionCache(
+        q=>({
+            invoke: ()=>{
+                let threshold = 0.3;
+                if (q.allData) {
+                    threshold = 0;
+                }
+                const dataQueryFilter = this.props.studyToDataQueryFilter[q.molecularProfile.studyId];
+                if (dataQueryFilter) {
+                    return internalClient.fetchCoExpressionsUsingPOST({
+                        molecularProfileId: q.molecularProfile.molecularProfileId,
+                        coExpressionFilter: dataQueryFilter as CoExpressionFilter,
+                        entrezGeneId: q.entrezGeneId,
+                        threshold
+                    });
+                } else {
+                    return Promise.resolve([]);
+                }
+            }
+        }),
+        q=>`${q.entrezGeneId},${q.molecularProfile.molecularProfileId}`
+    );
+
+    private get dataSetSelector() {
+        if (this.selectedMolecularProfile) {
+            let options = this.profiles.map(profile=>{
+                return {
+                    label: profile.name,
+                    value: profile.molecularProfileId
+                };
+            });
+            return (
+                <div
+                    style = {{
+                        display: "flex",
+                        alignItems: "center"
+                    }}
+                >
+                    <span>Data Set:</span>
+                    <div style={{display:"inline-block", width:300, marginLeft:4, marginRight:4}}>
+                        <Select
+                            name="data-set-select"
+                            value={this.selectedMolecularProfile.molecularProfileId}
+                            onChange={this.onSelectDataSet}
+                            options={options}
+                            searchable={false}
+                            clearable={false}
+                            className="coexpression-select-profile"
+                        />
+                    </div>
+                    <InfoIcon style={{marginLeft:3}} tooltip={<span>{this.selectedMolecularProfile.description}</span>}/>
+                </div>
+            );
+        } else {
+            return <LoadingIndicator isLoading={true}/>;
+        }
+    }
+
+    @bind
+    private header() {
+        return (
+            <div>
+                {this.dataSetSelector}
+            </div>
+        );
+    }
+
+    @bind
```

Comment:
how come these are bound?

Reply:
to prevent unnecessary react rerender. though now Im remembering you had said that Observer would deal with this.. are you sure about that?